src/enums.ts

```diff
@@ -129,3 +129,24 @@ export enum OnFailure {
   Continue = "continue_on_failure",
   Fail = "fail_the_build",
 }
+
+export enum HashAlgorithmEnum {
+  Unknown = "Unknown",
+  Md5 = "Md5",
+  Sha1 = "Sha1",
+  Sha256 = "Sha256",
+  Sha512 = "Sha512",
+}
+
+export enum HashEncodingEnum {
+  Utf8 = "utf8",
+  Base64 = "base64",
+  Binary = "binary",
+  Hex = "hex",
```

Comment:
should be uppercase for the API?
```suggestion
  Utf8 = "Utf8",
  Base64 = "Base64",
  Binary = "Binary",
  Hex = "Hex",
```